common/types/head_saver.go

```diff
@@ -0,0 +1,19 @@
+package types
+
+import (
+	"context"
+)
+
+// HeadSaver is an chain agnostic interface for saving and loading heads
+// Different chains will instantiate generic HeadSaver type with their native Head and BlockHash types.
+type HeadSaver[H Head[BLOCK_HASH], BLOCK_HASH Hashable] interface {
```

Comment:
HeadSaver will then be implemented only once, and I think it will only be used from within the headtracker package.
In that case, could we just move this to inside that package?

The common/types package should be only for types which need to be exported outside of the headtracker package.

Reply:
As discussed, this abstraction of subcomponents would be for a separate ticket

Reply:
https://chainlink-core.slack.com/archives/C04Q61M2MFT/p1684789327571249?thread_ts=1684463112.287119&cid=C04Q61M2MFT